[wiki] Document `yarn kbn watch` command for Kibana plugins (#6733)
elastic/eui: wiki/eui-team-processes/upgrading-kibana.md

node scripts/functional_test_runner/ --config x-pack/plugins/observability/e2e/synthetics_run.ts --bail --no-headless
```

### Debugging Kibana plugins

> Some EUI component usages are in `src/plugins/` as opposed to Kibana's `src/` or `x-pack/` folders (which hot reload automatically on page refresh). Kibana plugins **do not** update automatically unless you start a `yarn kbn watch` process in a new terminal alongside `yarn start`.

Reviewer: Thank you for adding this to the docs @cee-chen. These additions are how we incrementally improve DX.

I read through the change in the rendered file a few times and ran it through Hemingway app. What do you think about revising the first sentence to focus on where plugins reside? I look for code blocks, directory structures, and terminal commands when troubleshooting and don't filter information as well as I could. I'll admit 100% this suggestion is based on my tendency to skim when I'm in a hurry.

```suggestion
Some EUI component usages are in `src/plugins/`. Kibana plugins **do not** hot reload automatically. You must start a `yarn kbn watch` process in a new terminal, alongside `yarn start`.
```
